sonar-html-plugin/src/test/resources/checks/TableWithoutHeaderCheck.html

```diff
@@ -77,3 +77,48 @@
 <table aria-hidden="true">
   <table></table> <!-- Non-Compliant, no recursive check on parent -->
 </table>
+
+<table>
+  <caption>Some Table</caption>
+  <th:block th:insert="/ContainsTRAndTHAndTD.html" />
+</table>
+
+<table>
+  <caption>Some Table</caption>
+  <th:block th:replace="/ContainsTRAndTHAndTD.html" />
+</table>
+
+<table>
+  <caption>Some Table</caption>
+  <th:block th:include="/ContainsTRAndTHAndTD.html" />
+</table>
+
+<table th:include="/ContainsTRAndTHAndTD.html" >
+  <caption>Some Table</caption>
+</table>
+
+<table th:insert="/ContainsTRAndTHAndTD.html" >
+  <caption>Some Table</caption>
+</table>
+
+<table>
+  <div th:include="/blabla.html"></div> <!-- Non-Compliant : insertion will occur inside div -->
+</table>
+
+<table>
+  <div th:insert="/blabla.html"></div> <!-- Non-Compliant : insertion will occur inside div -->
+</table>
+
+<table>
+  <div th:replace="/ContainsTRAndTHAndTD.html"></div> <!-- Compliant : div will be replaced, maybe by <TH> -->
+</table>
+
+<table>
+  <div th:include="/blabla.html"></div>
+  <th:block th:replace="/ContainsTRAndTHAndTD.html" />
+</table>
+
+<table>
+  <th:block th:replace="/ContainsTRAndTHAndTD.html" />
+  <div th:include="/blabla.html"></div>
+</table>
```

Comment:
There's one case that remains to be covered: the case of potentially nested fragments :
```
<table>
  <tr>
    <div th:replace="/blabl.html"></div>
  </tr>
</table>
```

Reply:
I've added a test for that.